samples/java_layout/build.gradle

```diff
@@ -30,13 +41,29 @@ android {
 }
 
 dependencies {
+    // TODO Replace this with generic implementation for all sample SDKs to referencing remote versions is easier
+    def cioSDKVersion = localProperties["sdkVersion"]
+    if (cioSDKVersion == null || cioSDKVersion.toString().isBlank()) {
+        implementation(project(":sdk"))
+        implementation(project(":messagingpush"))
+        implementation(project(":messaginginapp"))
+    } else {
+        implementation "io.customer.android:tracking:$cioSDKVersion"
+        implementation "io.customer.android:messaging-push-fcm:$cioSDKVersion"
+        implementation "io.customer.android:messaging-in-app:$cioSDKVersion"
+    }
```

Comment:
For our sample apps, I don't think that we need to reference remote versions. 

The idea is: Do you want to test a specific version of the CIO SDK? Checkout that SDK tag/commit in git and then compile the sample apps to test that commit/tag. 

I don't see a reason we would need to test remote versions?

Reply:
I don't know what the ideal solution would be, but 
>The idea is: Do you want to test a specific version of the CIO SDK? Checkout that SDK tag/commit in git and then compile the sample apps to test that commit/tag.

this would always mean testing a local SDK, there should be a way to able to test code checked out from maven etc to be sure that works too. 

An example of it could be, what if while publishing artifacts, one fails? you would never realize that using local version all the time but could be caught when testing remote version

Reply:
+1 to what Shahroz said. And since this allows us to test both local and remote versions conveniently, we don't need to change anything here IMO.

Reply:
If the gradle configuration for the sample apps is: default to compiling the app against local SDK version but allow developer during development to install a remote version of SDK, then I am OK with this change. 

As part of the [QA CI workflow](https://github.com/customerio/issues/issues/9028), having sample apps compile on CI against local SDK for pull requests is important. Therefore, it should be the default behavior. Which, from looking at this gradle config file, that seems to be the case so I am OK with the configuration that exists here right now.